content/post/2019-06-18.md

```diff
@@ -0,0 +1,256 @@
+  { 
+   k=source_attribs[[j]]
+   raw_attrib<-table_csv[[i,k]]
+   cooked_attrib<-str_replace_all(raw_attrib,",","-") 
+   node_source_attribs[[j]]<-cooked_attrib
+  }
+  if (!(has.key(node_source, hash_nodes)))
+  {
+   hash_nodes[[node_source]] <- node_source_attribs
+   hash_links[[node_source]] <-0
+   hash_links_in[[node_source]] <-0
+   hash_links_out[[node_source]] <-0
+  }
+  else
+  {
+   node_source_attribs_old=hash_nodes[[node_source]]
+   if (identical(node_source_attribs_old,null_attrib)) 
+     {hash_nodes[[node_source]] <- node_source_attribs}
+  }
+  if (!(is.na(node_target)))
+  {
+   if (!(has.key(node_target, hash_nodes)))
+   {
+    hash_nodes[[node_target]] <- null_attrib
+    hash_links[[node_target]] <-0
+    hash_links_in[[node_target]] <-0
+    hash_links_out[[node_target]] <-0
+   }
+   par_nodes=paste(node_source,node_target)
+   if (!(has.key(par_nodes, hash_connections)))
+   {
+    hash_connections[[par_nodes]] <- 0
+    hash_connections_attrib[[par_nodes]]<-node_source_attribs
+   }
+   hash_connections[[par_nodes]] <- hash_connections[[par_nodes]] +1
+   hash_links[[node_source]] <- hash_links[[node_source]]+1
+   hash_links_out[[node_source]] <- hash_links_out[[node_source]]+1
+   hash_links[[node_target]] <-hash_links[[node_target]]+1
+   hash_links_in[[node_target]] <-hash_links_in[[node_target]]+1
+  }
+}
+```
+
+## Sort data according to connections
+
+The hash table object does not have the sort method, but has one to convert a hash table into a list. Once we have converted the hash list_links and list_connections into a list, we sort them down by number of connections.
+
+```r
+list_links=as.list.hash(hash_links )
+list_link_order = list_links[order(unlist(list_links), decreasing=TRUE)]
+list_connections=as.list.hash(hash_connections )
+list_connections_order = list_connections[order(unlist(list_connections), decreasing=TRUE)]
+```
+
+## Prepare the data for the GDF format
+
+In this step we place in GDF format nodes and links in descending order by number of connections.
+
+In the GDF format, the only data required for the definition of nodes is the name of the node, but attributes can be added. In this case, three fixed attributes are included, which are the total number of links, the number of inbound links and the number of outbound links. Since the GDF format is readable, these attributes allow getting an idea of the most relevant nodes even before importing them into Gephi. The attributes configured in the parameters are also added.
+
+The information of the nodes is stored in a matrix sized in rows by the number of nodes and in columns by the number of tributes configured plus four.
+
+For the definition of links only the source and target nodes are required, but we can also expand with attributes. In this case we add the weight of the relation, a boolean variable to indicate if the graph is directed or not (by default it is not directed) and the attributes configured in the parameters.
```

Comment:
En la primera frase:
... we can also expand THEM with attributes

Reply:
ok